sites/cnfpt/src/backend/base/cache.py

```diff
@@ -0,0 +1,209 @@
+        """
+        Instantiate the Redis Cache with server and params
+        and also the fallback cache accordingly to FALLBACK_BACKEND setting
+        """
+        super().__init__(params)
+        options = params.get("OPTIONS", {})
+        self._redis_cache = RedisCache(server, params)
+        self._fallback_cache = self.initialize_fallback_cache(
+            options.get(
+                "FALLBACK_BACKEND", "django.core.cache.backends.locmem.LocMemCache"
+            )
+        )
+
+    @staticmethod
+    def initialize_fallback_cache(backend, **kwargs):
+        """
+        Initialize fallback cache with the provided backend then return the
+        cache instance freshly created
+        """
+        fallback_cache = cache._create_cache(backend, **kwargs)
+        # Some caches -- python-memcached in particular -- need to do a cleanup at the
+        # end of a request cycle. If not implemented in a particular backend
+        # cache.close is a no-op. Not available in Django 1.5
+        if hasattr(fallback_cache, "close"):
+            signals.request_finished.connect(fallback_cache.close)
+
+        fallback_cache.set(INVALIDATION_FALLBACK_CACHE_KEY, False)
+        return fallback_cache
+
+    def _call_with_fallback(self, method, *args, **kwargs):
+        """
+        Try first to exec provided method through Redis cache instance,
+        in case of failure, logger reports the exception and
+        the fallback cache takes over.
+        """
+        try:
+            next_cache_state = self._call_redis_cache(method, args, kwargs)
+            self._invalidate_fallback_cache_if_needed()
+            return next_cache_state
+        except Exception as e:
+            self._fallback_cache.set(INVALIDATION_FALLBACK_CACHE_KEY, True)
+            logger.warning("[DEGRADED CACHE MODE] - Switch to fallback cache")
+            logger.exception(e)
+            return self._call_fallback_cache(method, args, kwargs)
+
+    def _call_redis_cache(self, method, args, kwargs):
+        """
+        Exec the provided method through the redis cache instance
+        """
+        return getattr(self._redis_cache, method)(*args, **kwargs)
+
+    def _call_fallback_cache(self, method, args, kwargs):
+        """
+        Exec the provided method through the fallback cache instance
+        """
+        return getattr(self._fallback_cache, method)(*args, **kwargs)
+
+    def _invalidate_fallback_cache_if_needed(self):
+        """
+        Check if fallback has been used, if yes invalidate its cms page cache.
+        """
+        if self._fallback_cache.get(INVALIDATION_FALLBACK_CACHE_KEY) is True:
+            version = self._fallback_cache.get(CMS_PAGE_CACHE_VERSION_KEY)
+            if version:
+                self._fallback_cache.set(
+                    CMS_PAGE_CACHE_VERSION_KEY,
+                    version,
+                    get_cms_setting("CACHE_DURATIONS")["content"],
+                )
```

Comment:
Instead of doing all this, maybe you could just you "incr"? 
https://redis.io/commands/incr

Reply:
> Increments the number stored at key by one. If the key does not exist, it is set to 0 before performing the operation.

Reply:
`incr` method do not accept a timeout parameter, so with this method I'm not sure that key lifetime is renewed.

https://github.com/django/django/blob/master/django/core/cache/backends/locmem.py#L69